### ISSUE TITLE
[BUG] setting pixel tolerance too high can break multipolygons

### ISSUE THREAD

**Issue by erik** [onthegomap/planetiler#496]
**Describe the bug**

Using a combination of `Feature.setPixelTolerance(0.5)` and `Feature.setMinPixelSize(0.1)` broke the geometry for [this relation](https://www.openstreetmap.org/relation/10041968), which caused the entire tile to be filled at z11.

<img width="1371" alt="Screen Shot 2023-02-22 at 10 21 17" src="https://user-images.githubusercontent.com/188935/220722035-693fed9e-0e22-47ba-b468-41b8c4fe3965.png">

If you look at the relation, the sand and grass polygons are super close to each other (but not touching), so it seems the combination of calls resulted in an invalid simplification.

<img width="1099" alt="grafik" src="https://user-images.githubusercontent.com/188935/220727798-356473dc-16f8-4f98-9a5c-70967a4e5710.png">

In my case, I tested that I can fix the tile by modifying the pixel size and tolerance values, but it seems possible for other multipolygons to have the same issue. The OpenMapTiles profile for example uses [the same sequence of calls](https://github.com/openmaptiles/planetiler-openmaptiles/blob/1cae8c4bdf99f66b616bdcd7d49342cb991a8d48/src/main/java/org/openmaptiles/layers/Building.java#L172-L173), and could conceivably run into this.

These changes all independently fixed the issue for me:

1. Remove the call to `setMinPixelSize`
2. Remove the call to `setPixelTolerance`
3. Change the pixel tolerance to `0.25`

**To Reproduce**

- GeoFabrik's `new-jersey` extract.
- Create a layer which selects `natural=sand` polygons.
- Use the following profile:

```java
    features
      .polygon("test")
      .setPixelTolerance(0.5)
      .setMinPixelSize(0.1)
      .setAttr("id", source.id())
```

- Relevant area: `11/40.9405/-74.0947`

Commenting out either of `setPixelTolerance` or `setMinPixelSize` will fix the issue

**Environment (please complete the following information):**

- Hardware: 2019 Macbook Pro
- OS: MacOS 12.13.1
- Java version and distribution: openjdk-17
- Maven version: 3.8.2

**Comment by msbarry** [comment]
Interesting, looks like it's right on the edge of the buffer of tile at zoom 11, here's the geometry it generates:

```
POLYGON ((-4 -4, 100.375 -4, 100.5 -3.875, 100.625 -4, 260 -4, 260 260, -4 260, -4 -4))
```

So it might be an issue with clipping... I've got it reproduced in a unit test, should be able to investigate from there.

**Comment by msbarry** [comment]
Put out a fix proposal in #499 - let me know if there are other cases you can think of to test there?
